CONTRIBUTING.md

```diff
@@ -0,0 +1,156 @@
+Please consider what kind of change it is:
+
+- For a **Major Feature**, first open an issue and outline your proposal so
+  that it can be discussed. This will also allow us to better coordinate our
+  efforts, prevent duplication of work, and help you to craft the change so
+  that it is successfully accepted into the project. For your issue name,
+  please prefix your proposal with `[discussion]`, for example "[discussion]:
+  your feature idea".
+- **Small Features** can be crafted and directly [submitted as a Pull
+  Request](#submit-pr).
+
+## <a name="submit"></a> Submission Guidelines
+
+### <a name="submit-issue"></a> Submitting an Issue
+
+Before you submit an issue, please search the issue tracker, maybe an issue for
+your problem already exists and the discussion might inform you of workarounds
+readily available.
+
+We want to fix all the issues as soon as possible, but before fixing a bug we
+need to reproduce and confirm it. In order to reproduce bugs we will
+systematically ask you to provide a minimal reproduction scenario using a
+repository or [Gist](https://gist.github.com/). Having a live, reproducible
+scenario gives us wealth of important information without going back & forth to
+you with additional questions like:
+
+- version of EquinoxJS used
+- 3rd-party libraries and their versions
+- and most importantly - a use-case that fails
+
+Unfortunately, we are not able to investigate / fix bugs without a minimal
+reproduction, so if we don't hear back from you we are going to close an issue
+that doesn't have enough info to be reproduced.
+
+You can file new issues [here][new_issue]
+
+### <a name="submit-pr"></a> Submitting a Pull Request (PR)
+
+Before you submit your Pull Request (PR) consider the following guidelines:
+
+1. Search [GitHub Pull Requests][gh_prs] for an open or closed PR
+   that relates to your submission. You don't want to duplicate effort.
+1. Fork this repository.
+1. Make your changes in a new git branch:
+
+   ```shell
+   git checkout -b my-fix-branch main
+   ```
+
+1. Create your patch, **including appropriate test cases**.
+1. Follow our [Coding Rules](#rules).
+1. Run the full test suite (see [common scripts](#common-scripts)),
+   and ensure that all tests pass.
+1. Commit your changes using a descriptive commit message.
+
+   ```shell
+   git commit -a
+   ```
+
+   Note: the optional commit `-a` command line option will automatically "add" and "rm" edited files.
+
+1. Push your branch to GitHub:
+
+   ```shell
+   git push origin my-fix-branch
+   ```
+
+1. In GitHub, send a pull request to `equinox-js:main`.
+
+- If we suggest changes then:
+
+  - Make the required updates.
+  - Re-run the test suites to ensure tests are still passing.
+  - Rebase your branch and force push to your GitHub repository (this will update your Pull Request):
+
+    ```shell
+    git rebase master -i
+    git push -f
+    ```
+
+That's it! Thank you for your contribution!
```

Comment:
Not sure what this adds. Might be more useful to convey the typical and worst case communication patterns - if you need the guidance in this section, you might not be ready for your effort to just sit there for N days/weeks due as sometimes can happen?

Reply:
Added some flavour